pkg/metrics/filteredlabels.go

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package metrics
+
+import (
+	"github.com/cilium/tetragon/pkg/metrics/consts"
+)
+
+type FilteredLabels interface {
```

Comment:
> What are filtered labels and why do we have to be an interface?

So the idea is that `FilteredLabels` represents common labels shared by multiple metrics, and that users can configure these (potentially high-cardinality) labels. It's an interface so that we can easily add common labels to metrics as a type parameter. It's not new, it was introduced in #2321. Currently there is only one practical type implementing it - `ProcessLabels` - but in the future I think we might define other sets of common labels, e.g. including "ip" for network metrics, or "kprobe" for low-level metrics.

> Which seems to mean that whenever we define a metric, we need to pass a type argument of FilteredLabels.

If a metric doesn't have common configurable labels, then we should use dedicated wrappers that use `NilLabels` - that's why they exist, so that we don't have to bother about `FilteredLabels` if not needed.